Updated readme with new TAGs (#1634)
cncf/toc: README.md

This is our [voting policy](docs/voting.md). Only TOC members can cast binding votes. (+1 Binding)

> ## Current Technical Advisory Groups

Reviewer: Should this be current or previous?

Author: Good point!

Author: What about "TAG being phased out" ?

Reviewer: sgtm!

Reviewer 2: Prefer just "Technical Advisory Groups" @riaankleinhans

> ## Technical Advisory Groups being phased out

Reviewer 3: very small nit, but we seem to use Title Case most places for headings in this document so this looks slightly off when reading through the rest of the doc?

> The TOC has approved the formation of [TAGs](tags/cncf-tags.md).

Reviewer 4: ```suggestion
The TOC has approved the formation of [TAGs](.archive/cncf-tags.md).
```

Author: Good catch, updated it.

* [TAG Observability](https://github.com/cncf/tag-observability)
* [TAG Environmental Sustainability](https://github.com/cncf/tag-env-sustainability)

> The current Technical Advisory Groups would be spun down during 2025 and replace with new Technical Advisory Groups and TOC Sub Projects, listed below.

Reviewer 3: ```suggestion
The current Technical Advisory Groups will be spun down during 2025 and replaced with new Technical Advisory Groups and TOC Sub Projects, listed below.
```

## New Technical Advisory Groups and TOC Sub Projects

### New Technical Advisory Groups

* [TAG Developer Experiance](./tags/tag-developer-experience/)
* [TAG Infrastructure](./tags/tag-infrastructure/)

> * [TAG Operational REsilience](./tags/tag-operational-resilience/)

Reviewer 3: ```suggestion
* [TAG Operational Resilience](./tags/tag-operational-resilience/)
```

Author: Updated, thanks

* [TAG Security and Compliance](./tags/tag-security-and-compliance/)
* [TG Workloads-foundation](./tags/tag-workloads-foundation/charter.md)

### TOC Sub Projects

> * [Contributor Strategy Subproject and Advocacy](./toc_subprojects/contributor-strategy-subproject-and-advocacy/)

Reviewer 2: nit: "SubProject" or "Sub Project" for consistency - same for Mentoring and Project Reviews

Reviewer 4: +1 to "SubProject". This makes consistency easier when appearing in file paths, e.g., `toc_subprojectas` vs. `toc_sub_projects`.

Reviewer 5: SubProject +1

Author: Updated to SubProject

* [Mentoring Subproject](./toc_subprojects/mentoring-subproject/)

> * [Project Reviews Subproject](./toc_subprojects/project-reviews-subproject/)

Reviewer 2: Please add "Artificial Intelligence Sub Project"

Reviewer 5: Aren't AI initiatives instead of subprojects?

Author: Artificial Intelligence Sub Project added, the link will not work as I have to update the tags.yaml in a next PR.

* [TAG Observability](https://github.com/cncf/tag-observability)
* [TAG Environmental Sustainability](https://github.com/cncf/tag-env-sustainability)

> The current Technical Advisory Groups would be spun down during 2025 and replace with new Technical Advisory Groups and TOC Sub Projects, listed below.

Reviewer 4: ```suggestion
The current Technical Advisory Groups would be spun down during 2025 and replace with new Technical Advisory Groups and TOC SubProjects, listed below.
```

Author: Updated, thanks

## New Technical Advisory Groups and TOC Sub Projects

### New Technical Advisory Groups

> * [TAG Developer Experiance](./tags/tag-developer-experience/)

Reviewer 4: ```suggestion
* [TAG Developer Experience](./tags/tag-developer-experience/)
```

Author: Updated, thanks

* [TAG Infrastructure](./tags/tag-infrastructure/)
* [TAG Operational REsilience](./tags/tag-operational-resilience/)
* [TAG Security and Compliance](./tags/tag-security-and-compliance/)

> * [TG Workloads-foundation](./tags/tag-workloads-foundation/charter.md)

Reviewer 4: ```suggestion
* [TAG Workloads Foundation](./tags/tag-workloads-foundation/charter.md)
```

Author: Updated, Thanks

> ### TOC Sub Projects

Reviewer 4: ```suggestion
### TOC SubProjects
```

Author: Updated, Thanks

* [Artificial Intelligence SubProject](./toc_subprojects/artificial-intelligence-subproject)
* [Contributor Strategy SubProject and Advocacy](./toc_subprojects/contributor-strategy-subproject-and-advocacy/)
* [Mentoring SubProject](./toc_subprojects/mentoring-subproject/)

> * [Project Reviews Sub Project](./toc_subprojects/project-reviews-subproject/)

Reviewer 4: ```suggestion
* [Project Reviews SubProject](./toc_subprojects/project-reviews-subproject/)
```

Author: Updated, thanks
